File: src/main/java/name/zeno/zervice/action/GoodsAction.java
package name.zeno.zervice.action;

import name.zeno.zervice.dao.entity.Goods;
import name.zeno.zervice.form.GoodsForm;
import name.zeno.zervice.logic.GoodsLogic;
import name.zeno.zervice.model.ApiResult;
import org.nutz.ioc.loader.annotation.Inject;
import org.nutz.ioc.loader.annotation.IocBean;
import org.nutz.mvc.annotation.*;

@IocBean
@At("/goods")
public class GoodsAction
{
    @Inject
    private GoodsLogic goodsLogic;

    @GET
    @At("/")
    public ApiResult<Goods> get()
    {
        ApiResult<Goods> result;

        result = new ApiResult<>();
        result.setData(goodsLogic.query(null));
        return result;
    }

    @POST
    @At("/")
    public ApiResult<Goods> create(@Param("param") GoodsForm form,
                                   @Param("param") String origin)
    {
        ApiResult<Goods> result = new ApiResult<>();
        result.setData(goodsLogic.create(form.getBody()));
        return result;
    }

    @PUT
    @At("/?")
    public ApiResult<Goods> update(long goodsId,
                                   @Param("param") GoodsForm form,
                                   @Param("param") String origin)
    {
        ApiResult<Goods> result = new ApiResult<>();
        Goods goods = form.getBody();
        goods.setGoodsId(goodsId);
        result.setData(goodsLogic.update(goods));
        return result;
    }

    @DELETE
    @At("/?")
    public ApiResult delete(long goodsId)
    {
        ApiResult result = new ApiResult();
        goodsLogic.delete(goodsId);
        return result;
    }
}

File: src/main/java/name/zeno/zervice/action/GoodsAction.kt
package name.zeno.zervice.action

import name.zeno.zervice.dao.entity.Goods
import name.zeno.zervice.form.GoodsForm
import name.zeno.zervice.logic.GoodsLogic
import name.zeno.zervice.model.ApiResult
import org.nutz.ioc.loader.annotation.Inject
import org.nutz.ioc.loader.annotation.IocBean
import org.nutz.mvc.annotation.*

@IocBean
@At("/goods")
class GoodsAction {
  @Inject
  lateinit private var goodsLogic: GoodsLogic

  @GET @At("/")
  fun get(): ApiResult<Goods> {
    val result: ApiResult<Goods>

    result = ApiResult<Goods>()
    result.setData(goodsLogic.query())
    return result
  }

  @POST @At("/")
  fun create(@Param("param") form: GoodsForm, @Param("param") origin: String)
      : ApiResult<Goods> {
    val result = ApiResult<Goods>()
    val goods = form.body ?: throw IllegalArgumentException("body == null")
    result.setData(goodsLogic.create(goods))
    return result
  }

  @PUT @At("/?")
  fun update(goodsId: Long,
             @Param("param") form: GoodsForm,
             @Param("param") origin: String): ApiResult<Goods> {
    val result = ApiResult<Goods>()
    val goods = form.body ?: throw IllegalArgumentException("body == null")
    goods.goodsId = goodsId
    result.setData(goodsLogic.update(goods))
    return result
  }

  @DELETE @At("/?")
  fun delete(goodsId: Long): ApiResult<*> {
    val result = ApiResult<Unit>()
    goodsLogic.delete(goodsId)
    return result
  }
}
